modules/gdnative/gdnative_api.json

```diff
@@ -2097,6 +2097,14 @@
           ["godot_pool_string_array *", "p_self"]
         ]
       },
+      {
+        "name": "godot_pool_string_array_join",
+        "return_type": "godot_string",
+        "arguments": [
+          ["godot_pool_string_array *", "p_self"],
+          ["const godot_string *", "p_delimiter"]
+        ]
+      },
```

Comment:
This would need to be defined in a new CORE API 1.3 here: https://github.com/godotengine/godot/blob/5dafc2cab1cad3d4022761619166e3502f63005e/modules/gdnative/gdnative_api.json#L20

Reply:
Thank you! Could you also check this PR: #55650 ? I guess I need to do the same there?..

Reply:
Done. I've updated `gdnative_api.json`. I've added new CORE API 1.3 and put newly added method within it.

Please, review :slightly_smiling_face: